src/game_objects/state.py

```diff
@@ -30,6 +30,16 @@ def reset(self, seed):
         self.bosses = []
         self.player_stats = {}
         self.guppy_set = set()
```

Comment:
There is a lot of code duplication ahead, here we should probably use a defaultdict of sets to gather them in one "transformations" dict.